Alembic

AlexPunches opened the pull request:
Добавил Алимбик и базовую миграцию базы и данных.
Добавил пункт про миграции в редми.

AlexPunches committed:
add env SQLALCHEMY_DATABASE_URL

AlexPunches committed:
Merge branch 'alchemy' of github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Praktikum into alchemy

AlexPunches committed:
make README.md for cross-platform

AlexPunches committed:
Merge branch 'alchemy' into develop

AlexPunches committed:
merge conflicts

AlexPunches committed:
Add Alembic

вместе с Алимбиком добавились стартовые миграции
в этих миграция две когорты в таблице Cohort

AlexPunches committed:
Merge branch 'develop' of github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Praktikum into develop

AntonZelinsky commented on README.md:
````diff
@@ -5,38 +5,47 @@ $ git clone git@github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Prak
-Запустить сервер
+4. Накатить миграции
+```bash
+$ alembic upgrade 3f3caaaf8457
````
alembic upgrade head

AntonZelinsky replied:
Так же, добавь команду для создания миграций

AntonZelinsky commented on README.md:
````diff
@@ -5,38 +5,47 @@ $ git clone git@github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Prak
 ```bash
 $ uvicorn app.main:app
 ```
 
````
Добавь ссылку на свой постман

AlexPunches replied:
постман надо как-то пограмотнее сделать, 
наверно, постепенно будем улучшать

AlexPunches committed:
merge develop

AntonZelinsky commented on alembic.ini:
```diff
@@ -0,0 +1,101 @@
+script_location = migrations
+
+# template used to generate migration files
+# file_template = %%(rev)s_%%(slug)s
```
Давай поменяем шаблон именования файлов на 
`file_template = %%(year)d-%%(month).2d-%%(day).2d-%%(hour).2d-%%(minute).2d-%%(second).2d_%%(rev)s_%%(slug)s`

AntonZelinsky replied:
Если так не сделать, то файлы будут иметь рандомные имена и не будет понятен порядок файлов

AlexPunches replied:
и правда :)

AntonZelinsky commented on migrations/versions/3f3caaaf8457_starting_data.py:
```diff
@@ -0,0 +1,58 @@
+    op.create_index(op.f('ix_Users_email'), 'Users', ['email'], unique=True)
+    op.create_index(op.f('ix_Users_id'), 'Users', ['id'], unique=False)
+    # ### end Alembic commands ###
+    op.bulk_insert(cohorts_table,
```
Сделай отдельную пустую миграцию для вставки этих данных. И назови её как-то соответсвующе

AntonZelinsky commented on migrations/versions/3f3caaaf8457_starting_data.py:
```diff
@@ -0,0 +1,58 @@
+"""Starting Data
+
+Revision ID: 3f3caaaf8457
```
Удали и заново создай начальную миграцию, чтобы в названии файла появилась дата и время создания его. Иначе будет неконсистенто. Про это я написал ниже

AntonZelinsky commented on README.md:
````diff
@@ -5,38 +5,47 @@ $ git clone git@github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Prak
+
+5. Запустить сервер
 ```bash
 $ uvicorn app.main:app
````
И добавь альтернативную команду для запуска `python run.py`

AlexPunches replied:
закралась опечатка в название EVENING_REMINDER_HOUR
поправил .env.example чтобы было как в коде

AlexPunches committed:
Fix migration and  README.md

справлена опечатка в .env.example
редми стал еще информативнее
разделил миграции на две, одна со структурой, другая с данными

AlexPunches committed:
fix Postmans link

AntonZelinsky commented on README.md:
```diff
@@ -5,46 +5,54 @@ $ git clone git@github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Prak
+[http://127.0.0.1:8000/docs](http://127.0.0.1:8000/docs)  
+
+### Важно!
+При изменении структуры базы не забывай накатывать мигации
```
`Создавать и применять`

AntonZelinsky commented on README.md:
````diff
@@ -5,46 +5,54 @@ $ git clone git@github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Prak
+$ python run.py
 ```
 
+## Тестировать
````
Какой-то уж сильно броский заголовок

AntonZelinsky commented on README.md:
````diff
@@ -5,46 +5,54 @@ $ git clone git@github.com:AntonZelinsky/ScheduleNotifierForMentorsOfYandex.Prak
 ```
 
-Запустить сервер
+4. Накатить миграции
````
Накатить — плохое слово для документации

AlexPunches replied:
я как-то не смог приличное слово подобрать :)

AlexPunches committed:
fix README.md
